cpp/src/io/json/json_column.cu

```diff
@@ -681,10 +667,10 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> device_json_co
         column_names.back().children = names;
       }
       auto [result_bitmask, null_count] = make_validity(json_col);
-      return {
-        make_structs_column(
-          num_rows, std::move(child_columns), null_count, std::move(result_bitmask), stream, mr),
-        column_names};
+      auto ret_col                      = make_structs_column(
+        num_rows, std::move(child_columns), cudf::UNKNOWN_NULL_COUNT, {}, stream, mr);
+      ret_col->set_null_mask(std::move(result_bitmask), null_count);
```

Comment:
```suggestion
      auto ret_col                      = make_structs_column(
        num_rows, std::move(child_columns), null_count, std::move(result_bitmask), stream, mr);
```
Can these 2 lines be combined?

Reply:
No. The code is changed to not pass null_mask to `make_structs_column`, is to skip the superimpose_nulls applied. null mask is set on the `cudf::column` object.

Reply:
I think a comment is needed here to explain why this creation is done in two steps. Otherwise we might have a "clean up" PR in six+ months that simplifies this code :D

How does `set_null_mask` impact null count?

Reply:
`set_null_mask` sets null count of `cudf::column` only. It does not call struct's superimpose_nulls.

Reply:
added comment explaining the reason.